project4/name_server/bootstrap.h

```diff
@@ -35,7 +35,7 @@ class Bootstrap : public Nameserver {
    * @param key The integer key for the corresponding value
    * @param val The value
    */
-  void Insert(uint key, std::string val);
+  void Insert(uint key, const std::string &val);
```

Comment:
`Nameserver` needs virtual destructor, and this class should override it. That might have been what `clang` was complaining about for the Mac build. Adding `~Bootstrap() override = default;` should do the trick.